### PR TITLE
Master fetch latest Silicon changes from upstream

### PR DESCRIPTION
**Pull request by PeterSuna**
- Change SchemaPath to Absolute
- Change SchemaContext to EffectiveModelContext
- Add YangParserException
- Set non-persistence data-store
- Use repackaged-akka artifact instead of Akka dependencies.

### CODE REVIEW

File: lighty-modules/lighty-openflow-sb/src/main/java/io/lighty/modules/southbound/openflow/impl/OpenflowSouthboundPlugin.java [hunk 1]
```diff
@@ -211,10 +215,18 @@ protected boolean initProcedure() {
...
+    @SuppressWarnings("checkstyle:illegalCatch")
     protected boolean stopProcedure() {
         destroy(this.packetListenerNotificationRegistration);
         destroy(this.flowCapableTopologyProvider);
         destroy(this.operationProcessor);
+        if (this.listenerRegistrationHelper != null) {
+            try {
+                listenerRegistrationHelper.close();
+            } catch (final Exception e) {
+                LOG.warn("Exception was thrown during closing listenerRegistrationHelper", e);
```
Comment (Pyro2266) [inline lighty-modules/lighty-openflow-sb/src/main/java/io/lighty/modules/southbound/openflow/impl/OpenflowSouthboundPlugin.java:227]: At the end of this method you should return `false` if anything failed.

File: lighty-modules/lighty-openflow-sb/src/main/resources/defaultOfpConfig.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
     "enableFlowRemovedNotification": true,
     "skipTableFeatures": true,
     "basicTimerDelay": 3000,
-    "maximumTimerDelay": 3679,
+    "maximumTimerDelay": 900000,
```
Comment (Pyro2266) [inline lighty-modules/lighty-openflow-sb/src/main/resources/defaultOfpConfig.json:10]: Is it OK to have maximumTimerDelay this big?

File: lighty-modules/lighty-openflow-sb/src/test/java/io/lighty/modules/southbound/openflow/tests/OpenflowSouthboundPluginTest.java [hunk 3]
```diff
@@ -8,11 +8,11 @@
 package io.lighty.modules.southbound.openflow.tests;
 
 import org.testng.Assert;
-import org.testng.annotations.Test;
 
 public class OpenflowSouthboundPluginTest extends OpenflowSouthboundPluginTestBase {
 
-    @Test(timeOut = 60_000)
+//    @Test(timeOut = 60_000)
+//    TODO: Allow when will be released OFP at version 0.12.0 or higher and serviceutils at 0.7.0 or higher
```
Comment (Pyro2266) [inline lighty-modules/lighty-openflow-sb/src/test/java/io/lighty/modules/southbound/openflow/tests/OpenflowSouthboundPluginTest.java:15]: With OFP snapshot versions this should be activated again

File: lighty-modules/lighty-openflow-sb/src/test/resources/ofpConfig.json [hunk 4]
```diff
@@ -6,7 +6,7 @@
     "basicTimerDelay": 2690,
     "echoReplyTimeout": 4000,
     "globalNotificationQuota": 9000,
-    "maximumTimerDelay": 3679,
+    "maximumTimerDelay": 900000,
```
Comment (Pyro2266) [inline lighty-modules/lighty-openflow-sb/src/test/resources/ofpConfig.json:9]: Is it OK to have maximumTimerDelay this big?

**Review by Pyro2266 — CHANGES_REQUESTED** [review]

File: lighty-modules/lighty-openflow-sb/src/main/resources/defaultOfpConfig.json (hunk 2 above)
Comment (PeterSuna) [reply lighty-modules/lighty-openflow-sb/src/main/resources/defaultOfpConfig.json:10]: Hmm yeah is harder to see it now. There was duplicate property with name nonZeroUint32Type. And this property was overriding maximumTimerDelay property in construction.
https://github.com/PANTHEONtech/lighty/pull/675/files/7521800919de874239c691bc48075cd41c24db9a#diff-b3861a128f6f9e37226de0209f6a20baf2e3aa8ceae31b82a472f3c8f7d6ec8a

So I removed that duplicate property with name nonZeroUint32Type.
```java
.setMaximumTimerDelay(new NonZeroUint32Type(this.nonZeroUint32Type))
``` 

But when "nonZeroUint32Type" property was overriding with his value the property "maximumTimerDelay" So I set "nonZeroUint32Type" value to  "maximumTimerDelay".

I don't know if this value is correct, but it was set like this before. But it was not seen.

File: lighty-core/dependency-versions/pom.xml [hunk 5]
```diff
@@ -20,12 +20,27 @@
         versions.
     </description>
 
+    <repositories>
+        <!-- TODO: Remove when will be available OFP ans Serviceutils release  -->
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply lighty-core/dependency-versions/pom.xml]: done

File: lighty-core/lighty-controller/src/test/java/io/lighty/core/controller/impl/tests/LightyControllerNotificationTest.java [hunk 6]
```diff
@@ -26,10 +27,10 @@ public void domNotificationServiceTest() throws InterruptedException, ExecutionE
         final LightyController lightyController = getLightyController();
 
         // setup
-        final SchemaPath schemaPath = SchemaPath.ROOT;
+        final Absolute schemaPath = Absolute.of(QName.create("namespace", "test"));
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply lighty-core/lighty-controller/src/test/java/io/lighty/core/controller/impl/tests/LightyControllerNotificationTest.java]: done

File: lighty-core/lighty-controller/src/test/java/io/lighty/core/controller/impl/tests/LightyControllerTestBase.java [hunk 7]
```diff
@@ -29,7 +29,7 @@
 
     private LightyController lightyController;
 
-    @BeforeClass(timeOut = 120_000)
+    @BeforeClass()
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply lighty-core/lighty-controller/src/test/java/io/lighty/core/controller/impl/tests/LightyControllerTestBase.java:32]: done

File: lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/NetconfCallhomePlugin.java [hunk 8]
```diff
@@ -27,8 +28,12 @@
     public NetconfCallhomePlugin(final LightyServices lightyServices, final String topologyId,
             final ExecutorService executorService, final AAAEncryptionService encryptionService) {
         super(executorService);
-        final DefaultBaseNetconfSchemas defaultBaseNetconfSchemas =
-                new DefaultBaseNetconfSchemas(lightyServices.getYangParserFactory());
+        final DefaultBaseNetconfSchemas defaultBaseNetconfSchemas;
+        try {
+            defaultBaseNetconfSchemas = new DefaultBaseNetconfSchemas(lightyServices.getYangParserFactory());
+        } catch (YangParserException e) {
+            throw new RuntimeException(e);
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/NetconfCallhomePlugin.java:35]: done

File: lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/NetconfClusteredTopologyPlugin.java [hunk 9]
```diff
@@ -34,7 +37,8 @@
 
     public NetconfClusteredTopologyPlugin(final LightyServices lightyServices, final String topologyId,
             final NetconfClientDispatcher clientDispatcher, final Integer writeTxIdleTimeout,
-            final ExecutorService executorService, final AAAEncryptionService encryptionService) {
+            final ExecutorService executorService, final AAAEncryptionService encryptionService)
+            throws YangParserException {
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/NetconfClusteredTopologyPlugin.java]: done

File: lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/NetconfTopologyPlugin.java [hunk 10]
```diff
@@ -31,7 +33,7 @@
 
     NetconfTopologyPlugin(final LightyServices lightyServices, final String topologyId,
             final NetconfClientDispatcher clientDispatcher, final ExecutorService executorService,
-            final AAAEncryptionService encryptionService) {
+            final AAAEncryptionService encryptionService) throws YangParserException {
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/NetconfTopologyPlugin.java]: done

File: lighty-modules/lighty-netconf-sb/src/test/java/io/lighty/modules/southbound/netconf/tests/NetconfBaseServiceBaseTest.java [hunk 11]
```diff
@@ -52,10 +55,13 @@ public void beforeTest() {
         baseSchema = new DefaultBaseNetconfSchemas(new YangParserFactoryImpl()).getBaseSchema();
     }
 
-    static EffectiveModelContext getEffectiveModelContext(final List<YangModuleInfo> moduleInfos) {
-        ModuleInfoBackedContext moduleInfoBackedCntxt = ModuleInfoBackedContext.create();
-        moduleInfoBackedCntxt.registerModuleInfos(moduleInfos);
-        return moduleInfoBackedCntxt.getEffectiveModelContext();
+    static EffectiveModelContext getEffectiveModelContext(final List<YangModuleInfo> moduleInfos)
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply lighty-modules/lighty-netconf-sb/src/test/java/io/lighty/modules/southbound/netconf/tests/NetconfBaseServiceBaseTest.java]: I have rework only body of this method. But this method was used only in constructor in the same class. So I add private scope.

File: lighty-modules/lighty-openflow-sb/src/main/java/io/lighty/modules/southbound/openflow/impl/OpenflowSouthboundPlugin.java (hunk 1 above)
Comment (PeterSuna) [reply lighty-modules/lighty-openflow-sb/src/main/java/io/lighty/modules/southbound/openflow/impl/OpenflowSouthboundPlugin.java:231]: done

File: lighty-modules/lighty-openflow-sb/src/test/java/io/lighty/modules/southbound/openflow/tests/OpenflowSouthboundPluginTest.java (hunk 3 above)
Comment (PeterSuna) [reply lighty-modules/lighty-openflow-sb/src/test/java/io/lighty/modules/southbound/openflow/tests/OpenflowSouthboundPluginTest.java]: done

**Review by Pyro2266 — APPROVED** [review]

File: lighty-core/dependency-versions/pom.xml [hunk 12]
```diff
@@ -34,62 +49,85 @@
...
+                <version>6.0.5</version>
                 <type>pom</type>
                 <scope>import</scope>
             </dependency>
 
             <dependency>
                 <groupId>org.opendaylight.openflowplugin</groupId>
                 <artifactId>openflowplugin-artifacts</artifactId>
-                <version>0.11.2</version>
+                <!--TODO change to release version when will be available -->
```
Comment (vainyksi) [inline lighty-core/dependency-versions/pom.xml:111]: change to FIXME

File: lighty-core/dependency-versions/pom.xml [hunk 13]
```diff
@@ -34,62 +49,85 @@
...
+                <version>0.12.0-SNAPSHOT</version>
                 <type>pom</type>
                 <scope>import</scope>
             </dependency>
             <dependency>
                 <groupId>org.opendaylight.serviceutils</groupId>
                 <artifactId>serviceutils-artifacts</artifactId>
-                <version>0.6.2</version>
+                <!--TODO change to release version when will be available -->
```
Comment (vainyksi) [inline lighty-core/dependency-versions/pom.xml:119]: change to FIXME

File: lighty-modules/integration-tests/pom.xml [hunk 14]
```diff
@@ -37,68 +37,24 @@
...
-        </dependency>
         <dependency>
             <groupId>io.lighty.core</groupId>
             <artifactId>lighty-controller</artifactId>
         </dependency>
         <dependency>
-            <groupId>org.glassfish.jersey.inject</groupId>
-            <artifactId>jersey-hk2</artifactId>
+            <groupId>io.lighty.resources</groupId>
+            <artifactId>singlenode-configuration</artifactId>
```
Comment (vainyksi) [inline lighty-modules/integration-tests/pom.xml:46]: those two can have also scope=test

File: lighty-core/lighty-controller/src/main/java/io/lighty/core/controller/api/LightyServices.java [hunk 15]
```diff
@@ -77,9 +75,9 @@
 
     DistributedDataStoreInterface getOperationalDatastore();
 
-    DOMDataTreeShardingService getDOMDataTreeShardingService();
+    DistributedShardFactory getDOMDataTreeShardingService();
 
-    DOMDataTreeService getDOMDataTreeService();
+    DistributedShardFactory getDOMDataTreeService();
```
Comment (vainyksi) [inline lighty-core/lighty-controller/src/main/java/io/lighty/core/controller/api/LightyServices.java:80]: these should be deprecated, shouldn't they?

and most probably referencing `getDistributedShardFactory` as replacement?

File: lighty-core/lighty-controller/src/main/java/io/lighty/core/controller/api/LightyServices.java [hunk 16]
```diff
@@ -65,7 +63,7 @@
 
     ActorSystemProvider getActorSystemProvider();
 
-    SchemaContextProvider getSchemaContextProvider();
+    EffectiveModelContextProvider getEffectiveModelContextProvider();
```
Comment (vainyksi) [inline lighty-core/lighty-controller/src/main/java/io/lighty/core/controller/api/LightyServices.java:66]: we should want to deprecate the old method (`getSchemaContextProvider()`) referencing newly created (`getEffectiveModelContextProvider`)

File: lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/NetconfClusteredTopologyPlugin.java [hunk 17]
```diff
@@ -16,44 +16,67 @@
...
-    private final NetconfTopologyManager topology;
+    private static final Logger LOG = LoggerFactory.getLogger(NetconfClusteredTopologyPlugin.class);
+
     private final DOMMountPointService domMountPointService;
+    private final LightyServices lightyServices;
+    private final String topologyId;
+    private final NetconfClientDispatcher clientDispatcher;
+    private final Integer writeTxIdleTimeout;
+    private final AAAEncryptionService encryptionService;
+    private NetconfTopologyManager topology;
```
Comment (vainyksi) [inline lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/NetconfClusteredTopologyPlugin.java:45]: can be converted to local variable

File: lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/NetconfTopologyPlugin.java [hunk 18]
```diff
@@ -16,37 +16,57 @@
...
 
-    private final NetconfTopologyImpl topology;
+    private static final Logger LOG = LoggerFactory.getLogger(NetconfTopologyPlugin.class);
+
     private final DOMMountPointService domMountPointService;
+    private final String topologyId;
+    private final NetconfClientDispatcher clientDispatcher;
+    private final AAAEncryptionService encryptionService;
+    private final LightyServices lightyServices;
+    private NetconfTopologyImpl topology;
```
Comment (vainyksi) [inline lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/NetconfTopologyPlugin.java:40]: can be converted to local variable

File: lighty-core/dependency-versions/pom.xml [hunk 19]
```diff
@@ -20,12 +20,27 @@
         versions.
     </description>
 
+    <repositories>
+        <!-- TODO: Remove when the release of OFP and Serviceutils will be available -->
```
Comment (vainyksi) [inline lighty-core/dependency-versions/pom.xml:24]: change to FIXME

File: lighty-modules/lighty-openflow-sb/pom.xml [hunk 20]
```diff
@@ -11,11 +11,22 @@
     <groupId>io.lighty.modules</groupId>
     <artifactId>lighty-openflow-sb</artifactId>
 
+    <repositories>
+        <!-- TODO: Remove when the release of OFP and Serviceutils will be available  -->
```
Comment (vainyksi) [inline lighty-modules/lighty-openflow-sb/pom.xml:15]: change to FIXME

**Review by vainyksi — CHANGES_REQUESTED** [review]

File: lighty-core/lighty-controller/src/main/java/io/lighty/core/controller/api/LightyServices.java (hunk 16 above)
Comment (PeterSuna) [reply lighty-core/lighty-controller/src/main/java/io/lighty/core/controller/api/LightyServices.java:66]: SchemaContextProvider was completely removed from yangtools in Silicon. Should I create  deprecate getSchemaContextProvider() method with return value EffectiveModelContextProvider?

File: lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/NetconfClusteredTopologyPlugin.java (hunk 17 above)
Comment (PeterSuna) [reply lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/NetconfClusteredTopologyPlugin.java:45]: Nope, I don't think that will be a good idea to edit initProcedure() method. Do you see any other option how can be these convert to local variable?

File: lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/NetconfTopologyPlugin.java (hunk 18 above)
Comment (PeterSuna) [reply lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/NetconfTopologyPlugin.java:40]: Nope, I don't think that will be a good idea to edit initProcedure() method. Do you see any other option how can be these convert to local variable?

File: lighty-core/dependency-versions/pom.xml (hunk 12 above)
Comment (PeterSuna) [reply lighty-core/dependency-versions/pom.xml]: done

File: lighty-core/dependency-versions/pom.xml (hunk 13 above)
Comment (PeterSuna) [reply lighty-core/dependency-versions/pom.xml]: done

File: lighty-modules/integration-tests/pom.xml (hunk 14 above)
Comment (PeterSuna) [reply lighty-modules/integration-tests/pom.xml]: done

File: lighty-core/lighty-controller/src/main/java/io/lighty/core/controller/api/LightyServices.java (hunk 15 above)
Comment (PeterSuna) [reply lighty-core/lighty-controller/src/main/java/io/lighty/core/controller/api/LightyServices.java]: done

File: lighty-core/dependency-versions/pom.xml (hunk 19 above)
Comment (PeterSuna) [reply lighty-core/dependency-versions/pom.xml]: done

File: lighty-modules/lighty-openflow-sb/pom.xml (hunk 20 above)
Comment (PeterSuna) [reply lighty-modules/lighty-openflow-sb/pom.xml]: done

File: lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/NetconfTopologyPlugin.java (hunk 18 above)
Comment (vainyksi) [reply lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/NetconfTopologyPlugin.java:40]: I mean, only `topology` variable, after refactoring constructor that should not be needed as a field

File: lighty-core/lighty-controller/src/main/java/io/lighty/core/controller/api/LightyServices.java (hunk 16 above)
Comment (vainyksi) [reply lighty-core/lighty-controller/src/main/java/io/lighty/core/controller/api/LightyServices.java:68]: OK, let's just switch those, 
use just `EffectiveModelContextProvider getEffectiveModelContextProvider()` as you did, as far as it is major release change in upstream

File: lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/NetconfTopologyPlugin.java (hunk 18 above)
Comment (PeterSuna) [reply lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/NetconfTopologyPlugin.java]: done

**Review by vainyksi — APPROVED** [review]

**Pull request merged**